test/Atmos/EDMF/closures/entr_detr.jl

```diff
@@ -44,78 +44,43 @@ function entr_detr(
     N_up = n_updrafts(m.turbconv)
     ρ_inv = 1 / gm.ρ
     a_up_i = up[i].ρa * ρ_inv
+    if a_up_i<FT(0)#m.turbconv.subdomains.a_min
+        E_dyn = FT(0)
+        Δ_dyn = FT(0)
+        E_trb = FT(0)
+    else
+        lim_E = entr.lim_ϵ
+        lim_amp = entr.lim_amp
+        w_min = entr.w_min
+        # precompute vars
+        w_up_i = up[i].ρaw / up[i].ρa
+        sqrt_tke = sqrt(max(en.ρatke, 0) * ρ_inv / env.a)
+        # ensure far from zero
+        Δw = filter_w(w_up_i - env.w, w_min)
+        w_up_i = filter_w(w_up_i, w_min)
+        Δb = up_aux[i].buoyancy - en_aux.buoyancy
+        D_E, D_δ, M_δ, M_E =
+            nondimensional_exchange_functions(m, entr, state, aux, t, ts, env, i)
 
-    lim_ϵ = entr.lim_ϵ
-    lim_amp = entr.lim_amp
-    w_min = entr.w_min
-    # precompute vars
-    w_up_i = up[i].ρaw / up[i].ρa
-    sqrt_tke = sqrt(max(en.ρatke, 0) * ρ_inv / env.a)
-    # ensure far from zero
-    Δw = filter_w(w_up_i - env.w, w_min)
-    w_up_i = filter_w(w_up_i, w_min)
+        # I am commenting this out for now, to make sure there is no slowdown here
+        Λ_w = abs(Δb / Δw)
+        Λ_tke = entr.c_λ * abs(Δb / (max(en.ρatke * ρ_inv, 0) + w_min))
+        λ = lamb_smooth_minimum(
+            SVector(Λ_w, Λ_tke),
+            m.turbconv.mix_len.smin_ub,
+            m.turbconv.mix_len.smin_rm,
+        )
 
-    Δb = up_aux[i].buoyancy - en_aux.buoyancy
+        # compute entrainment/detrainment components
+        E_trb =
+            2 * up[i].ρa * entr.c_t * sqrt_tke /
+            max(m.turbconv.pressure.H_up, entr.εt_min)
```

Comment:
This is something I don't understand fully either in the previous or current formulation. Why is there a `entr.εt_min`, but it does not have units of `εt`? Is this the minimum updraft height? If so, we should definitely change the name of this variable.

Reply:
it should be changed I will make a PR to change the name of this parameter